sympy/matrices/tests/test_matrices.py

```diff
@@ -2378,6 +2378,11 @@ def test_issue_5320():
         [0, 1, 0, 2]
     ])
 
+def test_issue_11944():
+    A = Matrix([[1]])
+    AIm = sympify(A)
+    assert Matrix.hstack(AIm, A) == Matrix([[1, 1]])
+    assert Matrix.vstack(AIm, A) == Matrix([[1], [1]])
```

Comment:
I'm not sure class should be preserved by stacking: in particular you mentioned mixed sparse and dense stacking.